src/reducers/ChatDux.js

```diff
@@ -7,50 +7,128 @@ const chat = createSlice({
   name: 'chat',
   initialState: {
     chatList: [],
-    chatRoom: [],
-    message: '',
-    chatRoomId: ''
+    chatRoom: {
+      chatRoomId: '',
+      sellerPrice: null,
+      sellerNumberOfShares: null,
+      buyerPrice: null,
+      buyerNumberOfShares: null,
+      updatedAt: null,
+      conversation: []
+    },
+    chatRoomId: '',
+    error: {
+      message: '',
+      statusCode: ''
+    }
   },
   reducers: {
-    updateChatListAction: (state, { payload }) => {
+    resChatRooms: (state, { payload }) => {
       // eslint-disable-next-line no-param-reassign
-      state.chatList = _orderBy(payload, ['createdAt'], ['desc']);
+      state.chatList = _orderBy(payload, ['updatedAt'], ['desc']);
     },
-    fetchChatListAction: () => {
-      SocketRequestService.requestChatList();
+    reqChatRooms: (state, { payload }) => {
+      SocketRequestService.requestChatRooms(payload);
     },
-    fetchChatRoomAction: (state, { payload }) => {
+    reqChatConversation: (state, { payload }) => {
       // eslint-disable-next-line no-param-reassign
       state.chatRoomId = payload.chatRoomId;
-      SocketRequestService.requestChatRoom({ chatRoomId: payload.chatRoomId });
+      SocketRequestService.requestChatConversation(payload);
     },
-    updateChatRoomAction: (state, { payload }) => {
+    resChatConversation: (state, { payload }) => {
       // eslint-disable-next-line no-param-reassign
       state.chatRoom = payload.chatRoom;
     },
-    fetchNewMessageAction: (state, { payload }) => {
+    reqNewMessage: (state, { payload }) => {
       SocketRequestService.requestNewMessage(payload);
     },
-    updateNewMessageAction: (state, { payload }) => {
-      state.chatRoom.push({ ...payload });
+    resNewChat: (state, { payload }) => {
+      const { newChat } = payload;
+      state.chatRoom.conversation.push({ ...newChat });
       const index = _findIndex(
         state.chatList,
         c => c.chatRoomId === payload.chatRoomId
       );
-      state.chatList.splice(index, 1, payload);
       // eslint-disable-next-line no-param-reassign
-      state.chatList = _orderBy(state.chatList, ['createdAt'], ['desc']);
+      state.chatList[index].updatedAt = payload.updatedAt;
+      // eslint-disable-next-line no-param-reassign
+      state.chatList = _orderBy(state.chatList, ['updatedAt'], ['desc']);
+    },
+    reqNewOffer: (state, { payload }) => {
```

Comment:
My plan was to have the an error message in the redux store. So whenever a request is made, the error message clears, then if there is an error, the error gets put in the redux store.

I used this method in my internship, so I thought this was the practice.

Reply:
I see. Can add a comment for that. However, I don't think we need to put errors in the redux store. If we do, we can always dispatch a `setError` action on catch instead of having these weird side effects in the same action.